### PR TITLE
Tuxedo application

### PR DESCRIPTION
**Pull request by JoshOrndorff**
#### Project Abstract

We propose to create Tuxedo, a framework for developing Substrate runtimes based on the Unspent Transaction Output (UTXO) model.

There are essentially two models for creating blockchain state machines:
* **Account System** as seen in Ethereum, Polkadot, and others
* **UTXO System** as seen in Bitcoin, Monero, Cardano, and others.

FRAME has been very successful in making the Accounts system accessible to Substrate developers. Tuxedo would make the UTXO model similarly accessible and begin to create a diversity of runtime frameworks, thus strengthening the entire Substrate Ecosystem.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (aUSD/USDT) address for the payment of the milestones is provided in the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — DISMISSED** [review]
Thanks for the application. @JoshOrndorff, I remember when you did a presentation on something about UTXO during the first sub0 event in Berlin 3 or 4 years ago. So when it comes to UTXO and Substrate, you are probably the most knowledgeable person, and I'm happy to support this application and share it with the rest of the team.

**Review by takahser — APPROVED** [review]
@JoshOrndorff super interesting idea, happy to approve this. This might allow Substrate to attract more Bitcoin-orientated people that are still sceptical of the Account-based model.

File: applications/tuxedo.md
Comment (semuelle) [inline applications/tuxedo.md:171]:
```suggestion
| **0e.** | Full written and Video Tutorial | With a node template, piece template, and user-facing wallet now complete, we can get serious about user and developer documentation. We will create a full written tutorial and video walkthough that covers how to build and run the Tuxedo Node Template, and send tokens around with the wallet. We will then dive into how to add the multisig piece to your runtime, and how to develop your own simple piece starting from the piece template. |
| 1. | Piece Template | We will create the template Tuxedo Piece analogous to the FRAME pallet template. This will allow runtime developers to have a concrete starting place when building their own utxo based Substrate runtimes. |
```

File: applications/tuxedo.md
Comment (semuelle) [inline applications/tuxedo.md:153]: Could you add the tech/language you would be using for this?

**Review by semuelle — CHANGES_REQUESTED** [review]
Just two minor requests. Otherwise exciting.

> a UTXO based smart contract language based on the pi calculus

Let us know when you get to that.

File: applications/tuxedo.md
Comment (JoshOrndorff) [reply applications/tuxedo.md]: I've added a bit about this. My vision is that the CLI will be written in Rust and communicate directly with the Substrate jsonrpsee endpoint. We won't get much value from projects like [subxt](https://github.com/paritytech/subxt) or [capi](https://github.com/paritytech/capi) because they all assume FRAME and its corresponding metadata.

**Comment by JoshOrndorff** [comment]
> @JoshOrndorff dismissed stale reviews from @takahser and @Noc2 via [3c0b41c](https://github.com/w3f/Grants-Program/commit/3c0b41c1186bd6c1bb1f4a9388a57c643cd40d10)

Oh oops. I didn't mean to dismiss your reviews. That happened automatically. Could I get your approvals again :pray: Sorry about that.

File: applications/tuxedo.md
Comment (0xCaso) [inline applications/tuxedo.md:92]: You should actually put the address here ;)
```suggestion
- **Registered Address:** -
```

Comment (JoshOrndorff) [reply applications/tuxedo.md:92]: I wondered if that's what you meant. I'm supposed to publish my physical location on the Earth for everyone to see? I'd prefer not to if that's at all possible. I see that the FuzzLand application put N/A. Is that acceptable?

Comment (Noc2) [reply applications/tuxedo.md:92]: That's fine. We only need the address as part of the invoice (which is not public).

**Review by Noc2 — APPROVED** [review]
Still happy to go ahead with it

**Comment by bernardoaraujor** [comment]
About a year ago I was conducting the Technical Feedback Process for the Ecosystem Success team at Parity.

While talking with Enjin/Efinity, @Lohann relayed to me that they could greatly benefit from a native UTxO support on Substrate.  

Gaming apps impose very high throughput requirements, they experienced difficulty synchronizing concurrent transactions, especially when game specific actions needed to be executed sequentially.

On account-based ledgers, every failed operation is reverted. For the gaming application, that imposes the requirement of a fallback logic that greatly increases its complexity. 

As a consequence, dealing with failed concurrent transactions became a highly consequential optimization issue in their pipeline.

An UTxO-based ledger that does not impose the need to process reverted transactions would help them achieve their desired throughput without overbloating the gaming apps with fallback logic for failed txs.

**Comment by NukeManDan** [comment]
In my limited knowlege of them, I think the context of zero knowledge proving systems, UTXOs will be far more reasonable and opimizable to use vs. account bases systems.

This work may prove foundationally critical to rollups and privacy tools as well. 

Will share with the [ZKP on Polkadot chat](https://matrix.to/#/!KCZvlFoofDQPUHhwIO:matrix.org?via=parity.io&via=web3.foundation) to get some others commenting on that front.

**Review by SBalaguer — APPROVED** [review]
I'm happy to go ahead with this! Bringing more flexibility to cater different project needs is very interesting.

**Review by randombishop — APPROVED** [review]
LG

**Comment by JoshOrndorff** [comment]
Thanks for the support everyone :pray:

We plan to **start work on Monday 13 Feb**. We're still wrapping up the latest academy right now.

**Pull request merged**
